### ISSUE TITLE
Add collaboration for sharing playlist items

### ISSUE THREAD

**Issue by quozl** [sugarlabs/jukebox-activity#22]
At the moment, Jukebox does not support collaboration.

Add collaboration.  When two or more Jukebox instances are shared;
* the playlist items are held in common, such that actions against the playlist also occur on the other instances,
* the Journal objects in the playlist (the audio or video files) will be transferred to the other instances,
* the audio and video players shall be synchronised, such that actions on one instance also occur on the other instances,
* actions that require completed transfer of the Journal object will be disabled until the transfer has completed.

**Comment by manaswinidas** [comment]
I'm interested to fix this issue. How do I get started?

**Comment by quozl** [comment]
Hello, and welcome.  I'm not familiar with your experience, so I may say more than I need to;

* set up two test environments of Sugar on a network and confirm you can [collaborate](https://github.com/sugarlabs/help-activity/blob/master/source/collaborating.rst) using the Chat activity,
* set up a [development environment](https://github.com/sugarlabs/sugar/blob/master/docs/development-environment.md) for Sugar activities,
* review all the source code in the https://github.com/sugarlabs/collabwrapper repository, and then test the sample activity in the `test` directory of that repository on the test environments,
* decide how to add CollabWrapper features to the Jukebox activity,
* produce a plan for user experience, and ask others to review,
* make changes to Jukebox, and ask others to review.

Ask any questions to clarify.

p.s. best place to ask or discuss general development process and community is on sugar-devel@ mailing list.  Not many people will contribute to a discussion inside an issue on GitHub.

**Comment by thegezy** [comment]
Hello, I will like to work on this issue.

**Comment by quozl** [comment]
Hello, and welcome.  I'm not familiar with your experience, so I may say more than I need to;

* set up two test environments of Sugar on a network and confirm you can [collaborate](https://github.com/sugarlabs/help-activity/blob/master/source/collaborating.rst) using the Chat activity,
* set up a [development environment](https://github.com/sugarlabs/sugar/blob/master/docs/development-environment.md) for Sugar activities,
* review all the source code in the https://github.com/sugarlabs/collabwrapper repository, and then test the sample activity in the `test` directory of that repository on the test environments,
* decide how to add CollabWrapper features to the Jukebox activity,
* produce a plan for user experience, and ask others to review,
* make changes to Jukebox, and ask others to review.

Ask any questions to clarify.

p.s. best place to ask or discuss general development process and community is on sugar-devel@ mailing list.  Not many people will contribute to a discussion inside an issue on GitHub.

**Comment by nikkuAg** [comment]
Hi,
Can I work on this issue

**Comment by quozl** [comment]
Hello, and welcome.  I'm not familiar with your experience, so I may say more than I need to;

* set up two test environments of Sugar on a network and confirm you can [collaborate](https://github.com/sugarlabs/help-activity/blob/master/source/collaborating.rst) using the Chat activity,
* set up a [development environment](https://github.com/sugarlabs/sugar/blob/master/docs/development-environment.md) for Sugar activities,
* review all the source code in the https://github.com/sugarlabs/collabwrapper repository, and then test the sample activity in the `test` directory of that repository on the test environments,
* decide how to add CollabWrapper features to the Jukebox activity,
* produce a plan for user experience, and ask others to review,
* make changes to Jukebox, and ask others to review.

Ask any questions to clarify.

p.s. best place to ask or discuss general development process and community is on sugar-devel@ mailing list.  Not many people will contribute to a discussion inside an issue on GitHub.

**Comment by sarthak0527** [comment]
Hi I would like to work on this issue.

**Comment by daredevil3435** [comment]
Hi, I would like to work. Can you explain me more? I'm newbie here

**Comment by quozl** [comment]
Sorry, no, it is not a suitable issue for a newbie.  You're welcome to prove me wrong with a pull request.

**Comment by B1TSH3LL** [comment]
I want to work on this issue

**Comment by dreamcatcher553** [comment]
I want to work on this issue

**Comment by Dimi20cen** [comment]
Wouldn't the transfer take very long in the case that users have a lot of files, escpecially video files? Would it be a good idea if instead of transfering the audio/video files to only show the title of the files to the other instances and give them the option to transfer only what they wish to collaboratively use?

**Comment by quozl** [comment]
Yes.  It would be best to transfer the metadata only, leaving the data to when it is required for playing.
